### ISSUE TITLE
Test dependencies use test-internal

### ISSUE THREAD

**Issue by OndrejSpanel** [scalacenter/sbt-dependency-submission#134]
I am not sure if this is a behavior which should be fixed or document, but when trying to use `configs-ignore: test`, I was surprised my test dependencies were still reported. When I have displayed the snapshot using `outputs.snapshot-json-path`, I see many (all?) test dependencies use `test-internal` as a config.

See https://github.com/OpenGrabeso/esprima-scala/actions/runs/6404384200/job/17384838363

```json
"resolved": {
        "org.scalatest:scalatest-shouldmatchers_sjs1_2.13:3.2.2": {
          "package_url": "pkg:maven/org.scalatest/scalatest-shouldmatchers_sjs1_2.13@3.2.2",
          "metadata": {
            "config": "test-internal"
          },
          "relationship": "indirect",
          "scope": "development",
          "dependencies": [
            "org.scalatest:scalatest-matchers-core_sjs1_2.13:3.2.2"
          ]
        },
```

I am not sure where this `test-internal` comes from, it is certainly not a configuration of my SBT project, which can be seen at https://github.com/OpenGrabeso/esprima-scala

**Comment by adpi2** [comment]
Thanks @OndrejSpanel for the report!

The `test-internal` configuration is a default sbt configuration. It is defined [here](https://github.com/sbt/librarymanagement/blame/aa090bdb251f83810ee17ba7ef0f813bd4754fc0/core/src/main/scala/sbt/librarymanagement/ConfigurationExtra.scala#L21) and it inherits from `Test`, `Provided` and `Optional`. It is used internally by sbt to compute the classpath of `Test` (we need the `Provided` and `Optional` dependencies to compile).

If you run `show update` in your build, you should see that there is indeed an `InternalTest` report.

```
[info] 	test:
[info] 		org.scala-lang:scala-library:2.13.12:default (): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.12/scala-library-2.13.12.jar), Map(info.apiURL -> https://www.scala-lang.org/api/2.13.12/), None, false),/home/piquerez/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.12/scala-library-2.13.12.jar)
[info] 		org.scala-lang.modules:scala-swing_2.13:2.1.1:default: (Artifact(scala-swing_2.13, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/modules/scala-swing_2.13/2.1.1/scala-swing_2.13-2.1.1.jar), Map(), None, false),/home/piquerez/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-swing_2.13/2.1.1/scala-swing_2.13-2.1.1.jar)
[info] 		com.fifesoft:rsyntaxtextarea:3.0.8:default: (Artifact(rsyntaxtextarea, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/com/fifesoft/rsyntaxtextarea/3.0.8/rsyntaxtextarea-3.0.8.jar), Map(), None, false),/home/piquerez/.cache/coursier/v1/https/repo1.maven.org/maven2/com/fifesoft/rsyntaxtextarea/3.0.8/rsyntaxtextarea-3.0.8.jar)
[info] 		org.scala-lang:scala-reflect:2.13.12:default (): (Artifact(scala-reflect, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.13.12/scala-reflect-2.13.12.jar), Map(info.apiURL -> https://www.scala-lang.org/api/2.13.12/scala-reflect/), None, false),/home/piquerez/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.13.12/scala-reflect-2.13.12.jar)
[info] 		com.github.opengrabeso:esprimascala_2.13:0.2.16: 
[info] 	test-internal:
[info] 		org.scala-lang:scala-library:2.13.12:default (): (Artifact(scala-library, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.12/scala-library-2.13.12.jar), Map(info.apiURL -> https://www.scala-lang.org/api/2.13.12/), None, false),/home/piquerez/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.12/scala-library-2.13.12.jar)
[info] 		org.scala-lang.modules:scala-swing_2.13:2.1.1:default: (Artifact(scala-swing_2.13, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/modules/scala-swing_2.13/2.1.1/scala-swing_2.13-2.1.1.jar), Map(), None, false),/home/piquerez/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-swing_2.13/2.1.1/scala-swing_2.13-2.1.1.jar)
[info] 		com.fifesoft:rsyntaxtextarea:3.0.8:default: (Artifact(rsyntaxtextarea, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/com/fifesoft/rsyntaxtextarea/3.0.8/rsyntaxtextarea-3.0.8.jar), Map(), None, false),/home/piquerez/.cache/coursier/v1/https/repo1.maven.org/maven2/com/fifesoft/rsyntaxtextarea/3.0.8/rsyntaxtextarea-3.0.8.jar)
[info] 		org.scala-lang:scala-reflect:2.13.12:default (): (Artifact(scala-reflect, jar, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.13.12/scala-reflect-2.13.12.jar), Map(info.apiURL -> https://www.scala-lang.org/api/2.13.12/scala-reflect/), None, false),/home/piquerez/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.13.12/scala-reflect-2.13.12.jar)
[info] 		com.github.opengrabeso:esprimascala_2.13:0.2.16:
```

There is an `internalConfigurationMap` key in sbt. We could use it to map the configuration to their internal configuration. From the point of view of the user, there would only be `test` but that would report the resolved dependencies of `internal-test`.

**Comment by OndrejSpanel** [comment]
> From the point of view of the user, there would only be `test` but that would report the resolved dependencies of `internal-test`

If that means `configs-ignore: test` will ignore those `test-internal` as well, I think this would be nice. From the user point of view I think `configs-ignore: test` does not do what is expected at the moment.

**Comment by github-nicolas-colomer** [comment]
Thanks for reporting this: we faced the exact same issue but had no idea where the problem come from.

We aws able to workaround by adding the `test-internal` configuration to the `configs-ignore` GitHub Action parameters:
```yaml
steps:
  # ...
  - uses: scalacenter/sbt-dependency-submission@v2
    with:
      # ...
      configs-ignore: test test-internal # see https://github.com/scalacenter/sbt-dependency-submission/issues/134
```

**Comment by adpi2** [comment]
I fixed this in [v2.3.1](https://github.com/scalacenter/sbt-dependency-submission/releases/tag/v2.3.1).

With `configs-ignore: test`, the `test-internal` configuration is also automatically ignored.
